rpcs3/Emu/Io/Skylander.h

```diff
@@ -4,18 +4,41 @@
 #include <queue>
 #include <mutex>
 
-struct sky_portal
+struct skylander
 {
-	std::mutex sky_mutex;
 	fs::file sky_file;
-	bool sky_reload          = false;
-	u8 sky_dump[0x40 * 0x10] = {};
+	u8 status = 0;
+	std::queue<u8> queued_status;
+	std::array<u8, 0x40 * 0x10> data{};
+	u32 last_id = 0;
+	void save();
+};
+
+class sky_portal
+{
+public:
+	void activate();
+	void deactivate();
+	void set_leds(u8 r, u8 g, u8 b);
+
+	void get_status(u8* buf);
+	void query_block(u8 sky_num, u8 block, u8* reply_buf);
+	void write_block(u8 sky_num, u8 block, const u8* to_write_buf, u8* reply_buf);
+
+	bool remove_skylander(u8 sky_num);
+	u8 load_skylander(u8* buf, fs::file in_file);
 
-	void sky_save();
-	void sky_load();
+protected:
+	std::mutex sky_mutex;
```

Comment:
Can you change it to shared_mutex since you are moving it anyway?

Reply:
Done.